modules/ble/src/bleio_seq_linux.c

```diff
@@ -197,6 +203,16 @@ void dec_ref_handle(BLEIO_SEQ_HANDLE_DATA* handle_data)
                     BUFFER_delete(instruction->data.buffer);
                 }
             }
+            /*
+                added by H. Ota
+            */
+			void* nextInstr = instruction->nextInst;
+			while (nextInstr!=NULL)
+			{
+				void* current = nextInstr;
+				nextInstr = ((BLEIO_SEQ_INSTRUCTION*)current)->nextInst;
+				free(current);
+			}
```

Comment:
To correctly free each BLE_SEQ_INSTRUCTION, we should delete the characteristic_uuid and the buffer (if it's a write instruction).  May I suggest replacing the contents of the for loop with:
```c
BLEIO_SEQ_INSTRUCTION *instruction = (BLEIO_SEQ_INSTRUCTION *)VECTOR_element(handle_data->instructions, i);
BLEIO_SEQ_INSTRUCTION *baseInstruction = instruction;
while (instruction != NULL)
{
    if (instruction->characteristic_uuid != NULL)
    {
        STRING_delete(instruction->characteristic_uuid);
    }

    if (is_write_instruction(instruction))
    {
        if (instruction->data.buffer != NULL)
        {
            BUFFER_delete(instruction->data.buffer);
        }
    }

    void* nextInstr = instruction->nextInst;
    if (instruction != baseInstruction)
    {
        free(instruction);
    }
    instruction = nextInstr;
}
```